contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -379,7 +379,8 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
     uint256 fallbackPrice;
     // 2nd word only read if stale
     uint96 minSpend;
-    // 3rd word only read during cancellation
+    uint8 decimals;
+    // 3rd word only read during cancellation and payment
```

Comment:
this is a great case for careful struct packing! Before this PR, only the 1st word is read on the hot path, but now we read the 1st and 3rd words on the hot path (which is probably partly responsible for the increase in gas overhead).

The first word is using 216 (32 + 24 + 160) bits, meaning there are still 40 (256 - 216) bits available! We totally have space in the first word for this field, and by putting it there, we can save an SLOAD :)

Reply:
Once again, I learnt the power of unit testing. after reordering the struct, gas is under control